modules/extension/app-schema/app-schema/src/main/java/org/geotools/data/complex/AppSchemaDataAccess.java

```diff
@@ -169,6 +179,57 @@ public AppSchemaDataAccess(Set<FeatureTypeMapping> mappings, boolean hidden)
         register();
     }
 
+    /**
+     * Checks if the mapping is an include and is identical to the previous one.
+     *
+     * @param mapping the mapping to check
+     * @return true if the mapping is an include and is identical to the previous one
+     */
+    private boolean isIncludeAndDeepSame(FeatureTypeMapping mapping) {
+        // Not an include, so no need to check
+        if (!Boolean.TRUE.equals(mapping.isInclude())) {
+            return false;
+        }
+        FeatureTypeMapping compareMapping = null;
+        // Get the mapping to compare with
+        if (mapping.getMappingName() != null) {
+            Name name = mapping.getMappingName();
+            if (this.mappings.containsKey(name)) {
+                compareMapping = this.mappings.get(name);
+            } else {
+                try {
+                    if (DataAccessRegistry.hasName(name)) {
+                        AppSchemaDataAccess appSchemaDataAccess =
+                                (AppSchemaDataAccess) DataAccessRegistry.getDataAccess(name);
+                        compareMapping = appSchemaDataAccess.getMappingByName(name);
+                    }
+                } catch (IOException | ClassCastException e) {
+                    LOGGER.fine(
+                            "Could not get mapping from registry for "
+                                    + name
+                                    + " while trying to test for duplicates");
+                    return false;
+                }
+            }
+            return mappingsDeepCompare(mapping, compareMapping);
+        }
+        return false;
+    }
+
+    /**
+     * Checks if the mappings are deep equal.
+     *
+     * @param mapping the mapping to compare
+     * @param compareMapping the mapping to compare with
+     * @return true if the mappings are deep equal
+     */
+    private boolean mappingsDeepCompare(
+            FeatureTypeMapping mapping, FeatureTypeMapping compareMapping) {
+        if (mapping == compareMapping) return true;
+        return Objects.equals(mapping.getAttributeMappings(), compareMapping.getAttributeMappings())
+                && Objects.equals(mapping.getMappingName(), compareMapping.getMappingName());
```

Comment:
The same comment applies here as above. So, it's not only the mapping name, which is optional, but we also need to check the target element name.

That said, should we also check for other properties of the mappings? Such as data sources, normalization, etc.? Would it make more sense to implement this equality check in the FeatureTypeMapping class itself?

Reply:
Target check added.  Data source id and normalization comparisons added.  Previously this was the equals method in FeatureTypeMapping but the accompanying hash method was causing test failures so I moved it here.

Reply:
Actually I just pushed a version with the logic moved to the equals method in FeatureTypeMapping and a corrected hash method and the tests are passing.